nav2_docking/opennav_docking/src/docking_server.cpp

```diff
@@ -284,15 +295,20 @@ void DockingServer::dockRobot()
     doInitialPerception(dock, dock_pose);
     RCLCPP_INFO(get_logger(), "Successful initial dock detection");
 
-    // Get the direction of the movement
+    // Get the direction of the movement and whether to drive backwards blindly
     bool dock_backward = dock_backwards_.has_value() ?
       dock_backwards_.value() :
       (dock->plugin->getDockDirection() == opennav_docking_core::DockDirection::BACKWARD);
+    bool backward_blind = dock->plugin->isBackwardBlind();
 
     // Docking control loop: while not docked, run controller
     rclcpp::Time dock_contact_time;
     while (rclcpp::ok()) {
       try {
+        // Perform pure rotation to dock orientation
+        if (backward_blind) {
```

Comment:
```suggestion
        if (dock->plugin->isBackwardBlind()) {
```

This is just a bool, we don't need `bool backward_blind = dock->plugin->isBackwardBlind();` stored beforehand